numpy/distutils/fcompiler/environment.py

```diff
@@ -70,11 +73,12 @@ def _get_var(self, name, conf_desc):
                 else:
                     var = envvar_contents
         if confvar is not None and self._conf:
-            var = self._conf.get(confvar, (None, var))[1]
-        if convert is not None:
-            var = convert(var)
+            if confvar in self._conf:
+                source, confvar_contents = self._conf[confvar]
+                var = convert(confvar_contents)
```

Comment:
Doesn't this potentially overwrite `var` set from `envvar`?

Reply:
If so, it should be documented. Looks like this is also not the only place that can happen. It would be useful to have a comment explaining the intended logic of this function.

Reply:
It does, but it always has done - this patch only affects the parsing, not the precedence of the various sources.

Reply:
Might be good to emit a warning if the overwrite happens

Reply:
I have the impression that conf file is higher priority than the environment variable -- is this documented anywhere?